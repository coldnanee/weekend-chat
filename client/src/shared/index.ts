export { SocketProvider, useSocketContext } from "./ui/socket/";
export { ReactQueryProvider } from "./lib/react-query";

export { DefaultAvatar } from "./ui/default-avatar";

export { Loader } from "./ui/loader";
export { createMetadata } from "./lib/createMetadata";
export { default } from "./lib/axios";
