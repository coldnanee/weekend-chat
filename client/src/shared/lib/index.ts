export { ReactQueryProvider } from "./react-query";
export { createMetadata } from "./createMetadata";
export { default } from "./axios";
export { getFormattedIsoDate } from "./getFormattedIsoDate";
export { refreshToken } from "./refreshToken";
export { i18n } from "./i18n";
