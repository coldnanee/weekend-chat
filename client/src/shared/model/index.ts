export { useSocketStore } from "./store";
export { refreshToken } from "./refreshToken";
