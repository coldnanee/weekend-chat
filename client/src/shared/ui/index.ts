export { Loader } from "./loader";
export { DefaultAvatar } from "./default-avatar";
export { SocketProvider } from "./socket";
export { SocketHandlers } from "./socket-handlers";
