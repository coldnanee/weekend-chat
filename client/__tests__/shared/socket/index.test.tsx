import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import { setCookie, deleteCookie } from "cookies-next";
import { SocketProvider, useSocketStore } from "@/shared";

describe("SocketProvider", () => {
	afterEach(() => {
		deleteCookie("accessJwt");
	});
	test("render", () => {
		render(<SocketProvider>CHILDREN</SocketProvider>);
		expect(screen.getByTestId("socket-provider")).toBeInTheDocument();
		expect(screen.getByTestId("socket-provider")).toMatchSnapshot();
	});

	test("check connection auth", () => {
		const mockConnect = jest.spyOn(useSocketStore.getState().socket, "connect");

		setCookie("accessJwt", "AUTH");
		render(<SocketProvider>CHILDREN</SocketProvider>);
		expect(mockConnect).toHaveBeenCalled();
	});
	test("check connection unauth", () => {
		const mockConnect = jest.spyOn(useSocketStore.getState().socket, "connect");

		render(<SocketProvider>CHILDREN</SocketProvider>);
		expect(mockConnect).toHaveBeenCalledTimes(0);
	});
});
