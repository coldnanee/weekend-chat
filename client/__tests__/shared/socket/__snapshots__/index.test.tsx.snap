// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocketProvider render 1`] = `
<div
  data-testid="socket-provider"
>
  CHILDREN
</div>
`;
